src/overlays/Overlay/index.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { useCallback, useEffect } from 'react';
+import styled from 'styled-components';
+
+import { Portal } from '../Portal';
+
+export interface OverlayProps {
+  children: React.ReactNode;
+  opened: boolean;
+  closeable?: boolean;
+  onClose?: () => void;
+  /** CSS z-index 속성의 값 */
+  zIndex?: number;
+}
+
+function useDefaultProps<T, DefaultProps extends Partial<T>>(props: T, defaultProps: DefaultProps): DefaultProps & T {
+  return {
+    ...defaultProps,
+    ...props,
+  };
+}
+
+export const Overlay = React.memo<OverlayProps>(originalProps => {
+  const { children, opened, closeable, onClose, zIndex } = useDefaultProps(originalProps, {
+    closeable: true,
+    zIndex: 3000,
+  });
+
+  useEffect(() => {
+    document.body.style.overflow = 'hidden';
```

Comment:
utils/ssr 에 isClient라는 함수가 있는데 그걸로 체크해주는게 좋을 것 같습니다
```suggestion
if (isClient()) {
    document.body.style.overflow = 'hidden';
```